sklearn/meson.build

```diff
@@ -18,6 +18,38 @@ if is_mingw
   add_project_arguments('-mlong-double-64', language: 'c')
 endif
 
+if not py.version().version_compare('>=3.9')
+  error('scikit-learn requires Python>=3.9, got ' + py.version() + ' instead')
+endif
```

Comment:
I tested and meson raises an error in that case

Reply:
I could remove it, but I like the fact that you have an early error if you have a too old Python.

I tested it locally and for now you need to install numpy and scipy to be told "actually you know what your Python is too old, sorry".

It seems it is `meson-python` that raises the error according to the error: 
```
  ('\x1b[31m',)meson-python: error: Package requires Python version >=3.9, running on 3.8.18
```

As noted in https://github.com/scikit-learn/scikit-learn/pull/28721#discussion_r1547468541 it would be nice if you can get the error without going through `pip`, e.g. `spin build` (if we use it one day) will call Meson directly.